jupyterhub/templates/_helpers-names.tpl

```diff
@@ -76,6 +76,11 @@
     {{- include "jupyterhub.fullname.dash" . }}hub
 {{- end }}
 
+{{- /* hub-serviceaccount ServiceAccount */}}
+{{- define "jupyterhub.hub-serviceaccount.fullname" -}}
+    {{- .Values.hub.serviceAccount.name | default (include "jupyterhub.hub.fullname" .) }}
```

Comment:
```suggestion
{{- if .Values.hub.serviceAccount.create -}}
    {{ default (include "jupyterhub.hub.fullname" .) .Values.hub.serviceAccount.name }}
{{- else -}}
    {{ default "default" .Values.hub.serviceAccount.name }}
{{- end -}}
```

As it is done in helm boilerplate?

Reply:
This change includes two parts of relevance:
1. serviceAccountName is always set, even when `.create` is false and `.name` is unspecified.
2. Style change, writing `default <default value> <primary value>` instead of `<primary value> | default <default value>`.
3. Style change, whitespace chomping practices

I don't want to accept the style changes as they are consistent across the Helm chart in a manner, but I think its acceptable to agree on the change 1 as it reduces the complexity of our template logic, making it easier to read.

@dingobar I'll let you decide about 1 as you also would be the person to do the work and I find both options fine.

Reply:
Hmmm, I'm inclined to agree, favoring implementing part 1 here.

Reply:
While I see the original functionality good and easy to understand, I buy the argument that since the default helm-chart base also sets "default" as the serviceAccount in the pods we should also do it.